Worker.js

```diff
@@ -148,9 +111,30 @@ onmessage = async (event) => {
 										}
 									}});
 								}
+								if (decrypted.decryptionRestored) {
+									postDecryptStatusMessage({
+										event: {
+											name: "decryptionRestored",
+											data: {
+												id,
+												senderId,
+											},
+										},
+									});
+								}
 							},
-							(error)=>{
-								//TODO: handle errors
+							(error) => {
+								if (error.message === 'decryptFailed') {
+									postDecryptStatusMessage({
+										event: {
+											name: "decryptFailed",
```

Comment:
Nit and IMO: you can remove some copy pasta here by passing through the error message:
```
event: {
  name: error.message,
  ...
```